ttt.c

```diff
@@ -15,12 +15,19 @@ static int no_step[2] = {0, 0};
 static int step_forward[2] = {1, 2};
 static int step_backward[2] = {-1, -2};
 
-int move_record[N_GRIDS];
-int move_count = 0;
+static int *move_record = NULL;
+static int move_count = 0;
 
 void record_move(int move)
 {
-    if (move_count < N_GRIDS) {
+    if (move_count == 0)
+        move_record = malloc(sizeof(int));
+    else
+        move_record = realloc(move_record, sizeof(int) * (move_count + 1));
```

Comment:
Frequent invocation of `realloc` may impact the performance. You can present certain budget for recorded movements in advance.